openmls/src/extensions/trust_anchor_extension.rs

```diff
@@ -0,0 +1,48 @@
+use serde::{Deserialize, Serialize};
+use tls_codec::{TlsDeserialize, TlsSerialize, TlsSize};
+
+/// Each `PerDomainTrustAnchor` represents a specific identity domain which is expected
+/// and authorized to participate in the MLS group. It contains the domain name and
+/// the specific trust anchor used to validate identities for members in that domain.
+/// ```c
+/// // draft-mahy-mls-group-anchors-00
+/// struct {
+///     opaque domain_name<V>;
+///     CredentialType credential_type;
+///     select (Credential.credential_type) {
+///         case x509:
+///             Certificate chain<V>;
+///     };
+/// } PerDomainTrustAnchor;
+/// ```
+#[derive(
+    Debug, Clone, PartialEq, Eq, Serialize, Deserialize, TlsSerialize, TlsDeserialize, TlsSize,
+)]
+pub struct PerDomainTrustAnchor {
+    domain_name: Vec<u8>,
+    credential_type: AnchorCredentialType,
+}
+
+/// Extension data for the anchors
+pub type PerDomainTrustAnchorsExtension = Vec<PerDomainTrustAnchor>;
+
+impl PerDomainTrustAnchor {
+    /// Creates a new instance of a `PerDomainTrustAnchor`
+    pub fn new(domain_name: Vec<u8>, credential_type: AnchorCredentialType) -> Self {
+        Self {
+            domain_name,
+            credential_type,
+        }
+    }
+}
+
+/// Defines the type of cretential for the domain trust anchor.
+#[derive(
+    Debug, Clone, PartialEq, Eq, Serialize, Deserialize, TlsSerialize, TlsDeserialize, TlsSize,
+)]
+#[repr(u8)]
+pub enum AnchorCredentialType {
+    /// X509 certificate chain
+    #[tls_codec(discriminant = 1)]
```

Comment:
The discriminant for `CredentialType::X509` is 2.
Maybe you should use a NewType struct to delegate to the existing `CredentialType`